doc/create-a-custom-rule.md

```diff
@@ -0,0 +1,242 @@
+# Create a custom `remark-lint` rule
+
+This guide is part of [a step-by-step tutorial](https://dev.to/floroz/how-to-create-a-custom-lint-rule-for-markdown-and-mdx-using-remark-and-eslint-2jim), and will help you getting started to create your first linting plugin for `remark`.
+
+## Table of Contents
+
+*   [Set up the project](#set-up-the-project)
+*   [Set up remark](#set-up-remark)
+*   [The no-invalid-gif rule](#the-no-invalid-gif-rule)
+*   [Create the custom rule](#create-the-custom-rule)
+*   [Rule arguments](#rule-arguments)
+*   [Rule implementation](#rule-implementation)
+*   [Import the rule in your remark config](#import-the-rule-in-your-remark-config)
+*   [Apply the rule on the Markdown file](#apply-the-rule-on-the-markdown-file)
+
+## Set up the project
+
+Create a new folder and navigate inside it from your Terminal. For this example I will be using UNIX commands (macOS and Linux compatible).
+Now we can generate our `package.json`
+
+```bash
+$ mkdir my-custom-rule
+
+$ cd my-custom-rule
+
+$ yarn init -y
+```
+
+Now we can start installing our dependencies.
+
+```bash
+$ yarn add remark-lint remark-cli
+```
+
+*   [remark-lint](https://github.com/remarkjs/remark-lint): a plugin to lint markdown built on [remark](https://github.com/remarkjs/remark): (a markdown processor).
+*   [remark-cli](https://github.com/remarkjs/remark/tree/main/packages/remark-cli): remark CLI.
+
+Because we will be working with [ASTs](https://en.wikipedia.org/wiki/Abstract_syntax_tree), we will also need some utilities:
```

Comment:
```suggestion
We will also need some utilities:
```

The utilities are not necessarily needed for ASTs, so I suggest removing that here. You could add it somewhere else.